db/sg_replicate_cfg.go

```diff
@@ -272,17 +276,22 @@ func (rc *ReplicationConfig) Equals(compareToCfg *ReplicationConfig) (bool, erro
 	return bytes.Equal(currentBytes, compareToBytes), nil
 }
 
-// Redact returns the ReplicationCfg with auth credentials of the remote database
-// redacted. Credentials will be redacted from replication config and remote URL.
-func (r *ReplicationConfig) Redact() *ReplicationConfig {
+// Redacted returns the ReplicationCfg with password of the remote database redacted from
+// both replication config and remote URL, i.e., any password will be replaced with xxxxx.
+func (r *ReplicationConfig) Redacted() *ReplicationConfig {
 	config := *r
-	if config.Username != "" {
-		config.Username = "****"
-	}
 	if config.Password != "" {
-		config.Password = "****"
+		config.Password = "xxxxx"
+	}
+	remote, err := url.Parse(config.Remote)
+	if err != nil {
+		base.Warnf("Error redacting password on remote URL [%s], %v", config.Remote, err)
+		return &config
+	}
+	if _, has := remote.User.Password(); has {
+		remote.User = url.UserPassword(remote.User.Username(), "xxxxx")
```

Comment:
2 methods for doing the same thing. Should be able to use the regexp approach once moved to a global function.

Reply:
Makes sense, this has been refactored in latest revision.